backend/src/models/shared.rs

```diff
@@ -0,0 +1,111 @@
+use nutype::nutype;
+use serde::{Deserialize, Serialize};
+use std::{net::SocketAddr, time::Duration};
+use utoipa::{IntoParams, ToSchema};
+
+#[nutype(sanitize(trim, lowercase) validate(not_empty, regex = r"(https?)?(?::[\/]{2})?([^\:]+):(([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$)"))]
```

Comment:
We should support any string in hostnames. Example: `localhost:20000` or `node1.qoollo.com:20000`

Reply:
Such patterns are supported: regexr.com/7lvdb